File: android/app/openglesapp/app/src/main/java/com/example/jniapp/MainActivity.java
package com.example.jniapp;

import android.app.Activity;
import android.content.res.AssetManager;
import android.os.Bundle;

public class MainActivity extends Activity {
    static {
        System.loadLibrary("native-lib");
    }

    private MyGLSurfaceView glSurfaceView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        glSurfaceView = new MyGLSurfaceView(this);
        setContentView(glSurfaceView);

        // 将 AssetManager 传递给 native 层
        AssetManager assetManager = getAssets();
        glSurfaceView.setAssetManager(assetManager);
    }

    @Override
    protected void onResume() {
        super.onResume();
        glSurfaceView.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
        glSurfaceView.onPause();
    }
}

File: android/app/bazel/src/main/java/com/example/hellobazel/MainActivity.kt
package com.example.hellobazel

import android.app.Activity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView

class MainActivity : Activity() {

    private var clickCount = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val textView = findViewById<TextView>(R.id.textView)
        val button = findViewById<Button>(R.id.button)

        button.setOnClickListener {
            clickCount++
            textView.text = "Clicked $clickCount times!"
        }
    }
}
